internal/hardware/hardware.go

```diff
@@ -1,30 +1,56 @@
 package hardware
 
 import (
-	"github.com/project-flotta/flotta-operator/models"
+	"reflect"
+
 	"github.com/openshift/assisted-installer-agent/src/inventory"
 	"github.com/openshift/assisted-installer-agent/src/util"
+	"github.com/project-flotta/flotta-operator/models"
 )
 
 type Hardware struct {
+	Dependencies util.IDependencies
 }
 
 func (s *Hardware) GetHardwareInformation() (*models.HardwareInfo, error) {
-	d := util.NewDependencies("/")
-	cpu := inventory.GetCPU(d)
-	hostname := inventory.GetHostname(d)
-	systemVendor := inventory.GetVendor(d)
-	interfaces := inventory.GetInterfaces(d)
-
+	s.Init()
 	hardwareInfo := models.HardwareInfo{}
+	err := s.GetHardwareImmutableInformation(&hardwareInfo)
```

Comment:
We got all these tested on internal/heartbeat, maybe should we move the test to this package?

Reply:
You mean to add test for hardware to do unit test of GetHardwareInformation and GetHardwareImmutableInformation?

Reply:
yeah, and try to keep heartbeat hardware agnostic.

Reply:
done